src/watchdog.c

```diff
@@ -129,6 +135,8 @@ IMPLEMENT_MOCKABLE_FUNCTION(, void, watchdog_reset, WATCHDOG_HANDLE, watchdog)
     }
     else
     {
+        /* Codes_SRS_WATCHDOG_45_003: [ watchdog_reset shall wait until state is not EXPIRING. ]*/
+        (void)InterlockedHL_WaitForNotValue(&watchdog->state, WATCHDOG_EXPIRING, UINT32_MAX);
         /*Codes_SRS_WATCHDOG_42_032: [ watchdog_reset shall set the state of the watchdog to STOP. ]*/
         (void)interlocked_exchange(&watchdog->state, WATCHDOG_STOP);

```

Comment:
seems like something that needs a compare_exchange too (in a loop), because it could go to EXPIRING state between lines 139 and 141

Reply:
same below

Reply:
Based on these comments, I think I'll change watchdog to use sm. Changes to follow.